modules/apps/layout/layout-taglib/src/main/java/com/liferay/layout/taglib/internal/servlet/taglib/LayoutStructureCommonStylesCSSTopHeadDynamicInclude.java

```diff
@@ -0,0 +1,493 @@
+			String value = stylesJSONObject.getString(styleName);
+
+			cssRuleSB.append(
+				StringUtil.replace(
+					CommonStylesUtil.getCSSTemplate(styleName), "{value}",
+					_getStyleValue(
+						frontendTokensJSONObject, styledLayoutStructureItem,
+						styleName, value)));
+			cssRuleSB.append(StringPool.NEW_LINE);
+		}
+
+		cssRuleSB.append("}\n");
+
+		return cssRuleSB.toString();
+	}
+
+	private String _getStyleFromStyleBookEntry(
+		JSONObject frontendTokensJSONObject, String styleValue) {
+
+		JSONObject styleValueJSONObject =
+			frontendTokensJSONObject.getJSONObject(styleValue);
+
+		if (styleValueJSONObject == null) {
+			return styleValue;
+		}
+
+		String cssVariable = styleValueJSONObject.getString(
+			FrontendTokenMapping.TYPE_CSS_VARIABLE);
+
+		return "var(--" + cssVariable + ")";
+	}
+
+	private JSONObject _getStylesJSONObject(
+		JSONObject itemConfigJSONObject, ViewportSize viewportSize) {
+
+		if (Objects.equals(viewportSize, ViewportSize.DESKTOP)) {
+			return itemConfigJSONObject.getJSONObject("styles");
+		}
+
+		return Optional.ofNullable(
+			itemConfigJSONObject.getJSONObject(viewportSize.getViewportSizeId())
+		).map(
+			viewportJSONObject -> viewportJSONObject.getJSONObject("styles")
+		).orElse(
+			JSONFactoryUtil.createJSONObject()
+		);
+	}
+
+	private String _getStyleValue(
+		JSONObject frontendTokensJSONObject,
+		StyledLayoutStructureItem styledLayoutStructureItem, String styleName,
+		String value) {
+
+		if (styleName.startsWith("margin") || styleName.startsWith("padding")) {
+			StringBundler sb = new StringBundler(5);
+
+			sb.append("var(--spacer-");
+			sb.append(value);
+			sb.append(StringPool.COMMA);
+			sb.append(_spacings.get(value));
+			sb.append("rem)");
+
+			return sb.toString();
+		}
+
+		if (Objects.equals(styleName, "backgroundImage")) {
+			return "var(--lfr-background-image-" +
+				styledLayoutStructureItem.getItemId() +
+					StringPool.CLOSE_PARENTHESIS;
+		}
+
+		if (Objects.equals(styleName, "opacity")) {
+			return String.valueOf(GetterUtil.getInteger(value, 100) / 100.0);
+		}
+
+		return _getStyleFromStyleBookEntry(frontendTokensJSONObject, value);
+	}
+
+	private boolean _includeStyles(
+		StyledLayoutStructureItem styledLayoutStructureItem, String styleName,
+		String value, ViewportSize viewportSize) {
+
+		if (Validator.isNull(value) ||
+			(Objects.equals(
+				value, CommonStylesUtil.getDefaultStyleValue(styleName)) &&
+			 Objects.equals(viewportSize, ViewportSize.DESKTOP))) {
+
+			return false;
+		}
+
+		if (!(styledLayoutStructureItem instanceof
+				ContainerStyledLayoutStructureItem)) {
+
+			return true;
+		}
+
+		ContainerStyledLayoutStructureItem containerStyledLayoutStructureItem =
+			(ContainerStyledLayoutStructureItem)styledLayoutStructureItem;
+
+		if (!Objects.equals(
+				containerStyledLayoutStructureItem.getWidthType(), "fixed")) {
+
+			return true;
+		}
+
+		if (Objects.equals(styleName, "marginLeft") ||
```

Comment:
@victorg1991 @ruben-pulido porque no incluimos para justo estos valores?

Reply:
Es por el container fixed, que ya pone valores en el marginLeft y Right

Lo explico en el mensaje del commit: 
>  - If we are rendering a container and have the fixed widthType, we don't need to add the marginLeft and marginRight styles since that type already includes a value for the margins, which is `auto`